mllib-dal/src/main/native/oneapi/dal/HomogenTableImpl.cpp

```diff
@@ -424,3 +419,56 @@ Java_com_intel_oneapi_dal_table_HomogenTableImpl_cGetDoubleData(
     env->SetDoubleArrayRegion(newDoubleArray, 0, datasize, data);
     return newDoubleArray;
 }
+
+/*
+ * Class:     com_intel_oneapi_dal_table_HomogenTableImpl
+ * Method:    cEmptyTableInit
+ * Signature: ()J
+ */
+JNIEXPORT jlong JNICALL Java_com_intel_oneapi_dal_table_HomogenTableImpl_cEmptyTableInit
+  (JNIEnv *env, jobject) {
+      printf(" init empty HomogenTable \n");
+      homogen_table *h_table = new homogen_table();
+      homogenPtr tablePtr = std::make_shared<homogen_table>(*h_table);
```

Comment:
Line 431-432 can be merged into single one. 
std::make_shared is to new the object and make share_ptr at the same time

```suggestion
      homogenPtr tablePtr = std::make_shared<homogen_table>();
```